daemon/cmd/daemon_main.go

```diff
@@ -1988,6 +1988,23 @@ func (d *Daemon) instantiateAPI(swaggerSpec *server.Spec) *restapi.CiliumAPIAPI
 	// /bgp/peers
 	restAPI.BgpGetBgpPeersHandler = NewGetBGPHandler(d.bgpControlPlaneController)
 
+	msg := "Required API option %s is disabled. This may prevent Cilium from operating correctly"
+	hint := "Consider enabling this API in " + server.AdminEnableFlag
+	for _, requiredAPI := range []string{
+		"GetConfig",
+		"GetHealthz",
+		"PutEndpointID",
+		"DeleteEndpointID",
+		"PostIPAM",
+		"DeleteIPAMIP",
```

Comment:
I would suggest to either leave a comment in the code explaining why each one of these APIs are fundamental and / or leave the reasoning for it in the Warning log message so that users / admins know why it is important.

Also, let's say that the admin removed the liveness probe from the daemonset and therefore doesn't need the `GetHealthz` and so it gets administratively disabled. Should we still print a Warning every time Cilium starts?

Reply:
:+1: comments make sense, I can send a follow-up.

Warnings are not perfect, but this is exactly the sort of reason it's a warning and not an error or fatal message. This is a way to let the administrator know that they're running an unconventional setup, and if they know what they're doing and are happy with the consequences then they can ignore it. If you're that far off the beaten path, then I'm OK with having a warning until someone complains that it's a real issue.

Reply:
https://github.com/cilium/cilium/pull/25170